flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -231,25 +230,39 @@ private BufferBuilder toBufferBuilder(MemorySegment memorySegment) {
 
 	@Nullable
 	private MemorySegment requestMemorySegment(boolean isBlocking) throws InterruptedException, IOException {
+		if (!isBlocking) {
```

Comment:
Sorry for adding a bit more of the confussion, but I just spotted one more thing: `requestMemorySegment(boolean isBlocking)` is actually used only in two places, one blockingly and the other non blocking. I think we can get rid of this `if (!isBlocking)` condition by simply inlining this method.

Also it would also get rid of the ugly `try/catch(InterruptedException)` in the non blocking method.